### PR TITLE
Fix up links in ML folder

### PR DESCRIPTION
**Pull request by gewarren**
- replace redirected links
- replace site-relative links with file-relative links

Contributes to #15560

### CODE REVIEW

File: docs/machine-learning/how-does-mldotnet-work.md [hunk 1]
```diff
@@ -222,7 +222,7 @@ You can transform input data into predictions in bulk, or one input at a time. I
     var price = predEngine.Predict(size);
 ```
  
-The `CreatePredictionEngine()` method takes an input class and an output class. The field names and/or code attributes determine the names of the data columns used during model training and prediction. You can read about  [How to make a single prediction](./how-to-guides/single-predict-model-ml-net.md) in the How-to section.
+The `CreatePredictionEngine()` method takes an input class and an output class. The field names and/or code attributes determine the names of the data columns used during model training and prediction. You can read about  [How to make a single prediction](how-to-guides/machine-learning-model-predictions-ml-net.md) in the How-to section.
```
Comment (cjgronlund) [inline docs/machine-learning/how-does-mldotnet-work.md]:
```suggestion
The `CreatePredictionEngine()` method takes an input class and an output class. The field names and/or code attributes determine the names of the data columns used during model training and prediction. See [Make predictions with a trained model](how-to-guides/machine-learning-model-predictions-ml-net.md) in the How-to guides.
```

Comment (cjgronlund) [reply docs/machine-learning/how-does-mldotnet-work.md]: LInk text and the target page heading need to align.

**Review by cjgronlund — CHANGES_REQUESTED** [review]
Good catches! Your corrections pointed out other issues with the content. See below.

**Comment by gewarren** [comment]
@cjgronlund Thanks for the review. I've addressed all of your suggestions.

**Review by cjgronlund — APPROVED** [review]

**Pull request merged**